pkg/client/openstack/admin/client.go

```diff
@@ -376,6 +378,31 @@ func (c *adminClient) CreateStorageContainer(projectID, containerName, serviceUs
 	return err
 }
 
+func (c *adminClient) ExistsStorageContainer(projectID, containerName string) (bool, error) {
+	endpointURL, err := c.getPublicObjectStoreEndpointURL(projectID)
+	if err != nil {
+		return false, err
+	}
+
+	storageClient, err := openstack.NewObjectStorageV1(c.ProviderClient, gophercloud.EndpointOpts{})
+	if err != nil {
+		return false, err
+	}
+	storageClient.Endpoint = endpointURL
+
+	_, err = containers.Get(storageClient, containerName, containers.GetOpts{}).Extract()
+	if err != nil {
+		err404 := gophercloud.ErrDefault404{}
```

Comment:
I think this can be simplified using a type assertion as we do here:
https://github.com/sapcc/kubernikus/blob/master/test/e2e/pyrolisis_test.go#L133